### ISSUE TITLE
add ImageMagick dependency?

### ISSUE THREAD

**Issue by Cody-G** [JuliaImages/TestImages.jl#36]
It seems ImageMagick is a test dependency but not a project dependency, leading to this failure on Julia 0.7:

```julia
julia> using TestImages; testimage("cameraman")
Error encountered while loading "/home/cody/.julia/packages/TestImages/6bT9D/images/cameraman.tif".
Fatal error:
ERROR: ArgumentError: Package ImageMagick not found in current path:
- Run `Pkg.add("ImageMagick")` to install the ImageMagick package.

Stacktrace:
 [1] handle_error(::ArgumentError, ::FileIO.File{FileIO.DataFormat{:TIFF}}) at /home/cody/.julia/packages/FileIO/Y0Lre/src/error_handling.jl:80
 [2] handle_exceptions(::Array{Any,1}, ::String) at /home/cody/.julia/packages/FileIO/Y0Lre/src/error_handling.jl:75
 [3] #load#27(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::FileIO.File{FileIO.DataFormat{:TIFF}}) at /home/cody/.julia/packages/FileIO/Y0Lre/src/loadsave.jl:193
 [4] load(::FileIO.File{FileIO.DataFormat{:TIFF}}) at /home/cody/.julia/packages/FileIO/Y0Lre/src/loadsave.jl:172
 [5] #load#13(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::String) at /home/cody/.julia/packages/FileIO/Y0Lre/src/loadsave.jl:118
 [6] load at /home/cody/.julia/packages/FileIO/Y0Lre/src/loadsave.jl:118 [inlined]
 [7] macro expansion at ./logging.jl:317 [inlined]
 [8] testimage(::String) at /home/cody/.julia/packages/TestImages/6bT9D/src/TestImages.jl:75
 [9] top-level scope at none:0
```

I think we just need to add ImageMagick to REQUIRE, I'll ready a PR.

**Comment by timholy** [comment]
QuartzImageIO is used by default on OSX, so making it part of the required packages is not without issues.

FileIO used to offer to install ImageMagick for you, but that was deemed too magical. (I can understand why.)

**Comment by Cody-G** [comment]
Ah I see.  I ran into this when testing using TestImages on Travis.  It sounds like rather than adding `ImageMagick` as a test dependency it will be better to add a build script that checks whether the user is running linux / mac / windows in order to decide which library to install?

**Comment by timholy** [comment]
Given that it shows an informative error about how to fix the problem, do we actually need to do anything? Yes I'm being lazy, but it's laziness with a purpose :smile:. If we do nothing then FileIO is in charge of deciding who should load images; if a new package comes on line (e.g., https://github.com/c42f/OpenImageIO.jl) then FileIO is the one-and-only thing that needs updating. If we add things to the build script here then there is more stuff to change.

**Comment by Cody-G** [comment]
>If we add things to the build script here then there is more stuff to change.

This makes sense.  I wasn't very clear in my post, I was trying to decide on the best practice when testing a package that depends on TestImages.  Because currently adding TestImages as a dependency will cause a failure on Travis unless ImageMagick is also a dependency.  Since we wouldn't just want to install ImageMagick unconditionally, I thought to edit the build script to install it conditionally in this dependent package I'm working on.  Does that sound reasonable?

**Comment by timholy** [comment]
Got it, sorry, you were pretty clear and I was just being dense. How about this:
- For a package that has a `Project.toml` file, you can add it to the "extras" section and "test" target (or I think it's now possible to add a `Project.toml` file to the `test/` directory, and if present it will be used).
- Otherwise, add it to `test/REQUIRE`.

**Comment by musm** [comment]
bump. I just installed this package and tried
```julia
julia>  using Images, TestImages, ImageInTerminal
[ Info: Precompiling Images [916415d5-f1e6-5110-898d-aaa5f9f070e0]
[ Info: Precompiling TestImages [5e47fb64-e119-507b-a336-dd2b206d9990]
[ Info: Precompiling ImageInTerminal [d8c32880-2388-543b-8c61-d9f865259254]

julia> testimage("cameraman")
Error encountered while loading "C:\\Users\\Mus\\.julia\\packages\\TestImages\\6bT9D\\images\\cameraman.tif".
Fatal error:
ERROR: ArgumentError: Package ImageMagick not found in current path:
- Run `import Pkg; Pkg.add("ImageMagick")` to install the ImageMagick package.

```

**Comment by Cody-G** [comment]
Given the cross-platform difficulties described above I was convinced not to add ImageMagick as a dependency.  My followup question (how to add TestImages as test dependency of  another package) was also answered, though I never got the chance to test whether that solution works on all platforms.  I probably should have closed the issue, if others are still unsatisfied I'll leave it open.

**Comment by johnnychen94** [comment]
> I think it's now possible to add a Project.toml file to the test/ directory, and if present it will be used

@timholy can you find any reference to this?

**Comment by timholy** [comment]
https://julialang.github.io/Pkg.jl/dev/creating-packages/#Test-specific-dependencies-in-Julia-1.2-and-above-1

**Comment by johnnychen94** [comment]
🤔 Looks like we still can't solve the platform-dependent package deps using the new `Project.toml` even with this.

My current workaround is to add both `ImageMagick` and `QuartzImageIO` into `test` target. - The CI result says ok.

**Comment by timholy** [comment]
This has been fixed if you use Julia 1.3 or higher.
